### PR TITLE
UX-655, UX-674 Convert Tabs to typescript

### PR DESCRIPTION
**Pull request by jonambas**
### What Changed
- Converts `Tabs` to typescript
- Converts `useResizeObserver` to typescript

### How To Test or Verify
- Verify the `useResizeObserver` and `Tabs` stories function correctly

### PR Checklist

- [x] Add the correct `type` label
- [ ] Pull request approval from #uxfe or #design-guild

### CODE REVIEW

File: packages/matchbox/src/hooks/useResizeObserver.ts [hunk 1]
```diff
@@ -23,7 +23,7 @@ import ResizeObserver from 'resize-observer-polyfill';
  *  return <div ref={ref} />;
  * };
  */
-function useResizeObserver() {
+function useResizeObserver(): [React.Dispatch<any>, { contentRect?: { [k: string]: unknown } }] {
```
Comment (jonambas) [inline packages/matchbox/src/hooks/useResizeObserver.ts:26]: Not sure if i did this correctly

File: packages/matchbox/src/components/Tabs/Tab.tsx [hunk 2]
```diff
@@ -1,35 +1,36 @@
...
 import { tabStyles } from './styles';
-import { deprecate } from '../../helpers/propTypes';
+import * as Polymorphic from '../../helpers/types';
+
+export type TabProps = UnstyledLinkProps & {
+  selected?: number;
+  fitted?: boolean;
+  content?: React.ReactNode;
+  index?: number;
+  onClick?: (event: React.MouseEvent<HTMLAnchorElement, MouseEvent>, index: number) => void;
```
Comment (logansparlin) [inline packages/matchbox/src/components/Tabs/Tab.tsx:12]: I don't think we need the second `MouseEvent` parameter here but it's not hurting anything so your call

File: packages/matchbox/src/components/Tabs/useTabConstructor.tsx [hunk 3]
```diff
@@ -10,13 +23,21 @@ function useTabConstructor({
   onSelect,
   keyboardActivation,
   disableResponsiveBehavior,
-}) {
+}: {
+  tabs?: TabProps[];
+  fitted?: boolean;
+  selected?: number;
+  handleClick?: (event: React.MouseEvent<HTMLAnchorElement, MouseEvent>, index: number) => void;
```
Comment (logansparlin) [inline packages/matchbox/src/components/Tabs/useTabConstructor.tsx:30]: Same comment about the second `MouseEvent` param

**Review by logansparlin — APPROVED** [review]
looks good, just a couple small comments

**Pull request merged**